Mathlib/Analysis/Normed/Module/Completion.lean

```diff
@@ -66,46 +62,39 @@ theorem norm_toComplL {𝕜 E : Type*} [NontriviallyNormedField 𝕜] [NormedAdd
     [NormedSpace 𝕜 E] [Nontrivial E] : ‖(toComplL : E →L[𝕜] Completion E)‖ = 1 :=
   (toComplₗᵢ : E →ₗᵢ[𝕜] Completion E).norm_toContinuousLinearMap
 
+end Module
+
 section Algebra
 
-variable (𝕜) (A : Type*)
-
-instance [SeminormedRing A] : NormedRing (Completion A) :=
-  { Completion.ring,
-    Completion.instMetricSpace with
-    dist_eq := fun x y => by
-      refine Completion.induction_on₂ x y ?_ ?_ <;> clear x y
-      · refine isClosed_eq (Completion.uniformContinuous_extension₂ _).continuous ?_
-        exact Continuous.comp Completion.continuous_extension continuous_sub
-      · intro x y
-        rw [← Completion.coe_sub, norm_coe, Completion.dist_eq, dist_eq_norm]
-    norm_mul := fun x y => by
-      refine Completion.induction_on₂ x y ?_ ?_ <;> clear x y
-      · exact
-          isClosed_le (Continuous.comp continuous_norm continuous_mul)
-            (Continuous.comp _root_.continuous_mul
-              (Continuous.prod_map continuous_norm continuous_norm))
-      · intro x y
-        simp only [← coe_mul, norm_coe]
-        exact norm_mul_le x y }
-
-instance [SeminormedCommRing A] [NormedAlgebra 𝕜 A] [UniformContinuousConstSMul 𝕜 A] :
-    NormedAlgebra 𝕜 (Completion A) :=
-  { Completion.algebra A 𝕜 with
-    norm_smul_le := fun r x => by
-      refine Completion.induction_on x ?_ ?_ <;> clear x
-      · exact
-          isClosed_le (Continuous.comp continuous_norm (continuous_const_smul r))
-            (Continuous.comp (continuous_mul_left _) continuous_norm)
-      · intro x
-        simp only [← coe_smul, norm_coe]
-        exact norm_smul_le r x }
+variable (A : Type*)
+
+instance [SeminormedRing A] : NormedRing (Completion A) where
+  __ : NormedAddCommGroup (Completion A) := inferInstance
+  __ : Ring (Completion A) := inferInstance
+  norm_mul x y := by
+    induction x, y using induction_on₂ with
+    | hp =>
+      exact
+        isClosed_le (Continuous.comp continuous_norm continuous_mul)
+          (Continuous.comp _root_.continuous_mul
+            (Continuous.prod_map continuous_norm continuous_norm))
+    | ih x y =>
+      simp only [← coe_mul, norm_coe]
+      exact norm_mul_le x y
+
+instance [SeminormedCommRing A] : NormedCommRing (Completion A) where
+  __ : CommRing (Completion A) := inferInstance
+  __ : NormedRing (Completion A) := inferInstance
+
+instance
+    [NormedField 𝕜] [SeminormedCommRing A] [NormedAlgebra 𝕜 A] [UniformContinuousConstSMul 𝕜 A] :
+    NormedAlgebra 𝕜 (Completion A) where
```

Comment:
the bare `instance` line bugs me stylistically. Care to reformat?

Reply:
Would you be hainstance ppy with `instNormedAlgebra`?

Reply:
It turns out that this last argument was redundant anyway.